### PR TITLE
Cext 3114

### PR DESCRIPTION
**Pull request by nvyasadobe**
Secrets file validation

## Description

This PR is related to secrets file validation in windows and unix environment.

## Related Issue

https://jira.corp.adobe.com/browse/CEXT-3114

## Motivation and Context

* Validate secrets file in various aspects and throw relevant messages.

## How Has This Been Tested?

Loaded the new CLI plugin locally using the `aio plugins link .` command against branch.
Tested all status combinations.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: src/utils.js [hunk 1]
```diff
@@ -427,9 +430,10 @@ async function validateSecretsFile(secretsFile) {
 async function interpolateSecrets(secretsFilePath, command) {
 	try {
 		const secretsContent = await readFileContents(secretsFilePath, command, 'secrets');
+
 		// Check if environment variables are used in the file content
-		if (os.platform() === 'win32' && /(\$[a-zA-Z_][a-zA-Z0-9_]*)/.test(secretsContent)) {
-			throw new Error('Batch variables are not supported in YAML files on Windows.');
+		if (os.platform() === 'win32' && /\$({)?[a-zA-Z_][a-zA-Z0-9_]*}?/.test(secretsContent)) {
```
Comment (revanth0212) [inline src/utils.js:435]: Just checking, did you add tests around this?

File: src/utils.js [hunk 2]
```diff
@@ -458,10 +462,37 @@ async function parseSecrets(secretsContent) {
...
+/**
+ * This function returns user friendly errors that occurs while YAML.parse
+ *
+ * @param error errors from YAML.parse
+ */
+function getSecretsYamlParseError(error) {
+	if (error.code === 'BAD_INDENT') {
+		return 'Bad indentation error occurred while parsing YAML data: ' + error.message;
+	} else if (error.code === 'DUPLICATE_KEY') {
+		return 'Duplicate Keys found while parsing YARN data: ' + error.message;
```
Comment (revanth0212) [inline src/utils.js:493]:
```suggestion
		return 'Invalid YAML - Found Duplicate Keys: ' + error.message;
```

File: src/utils.js [hunk 3]
```diff
@@ -458,10 +462,37 @@ async function parseSecrets(secretsContent) {
...
+	} else {
+		return 'An unexpected error occurred: ' + error.message;
```
Comment (revanth0212) [inline src/utils.js:495]:
```suggestion
		return 'Unexpected Error: ' + error.message;
```

**Review by revanth0212 — CHANGES_REQUESTED** [review]
Great work dude. Please verify the feedback comments. Just a couple simple ones.

File: src/utils.js (hunk 1 above)
Comment (nvyasadobe) [reply src/utils.js:435]: Done, have added tests specific to batch variables in [this commit](https://github.com/adobe/aio-cli-plugin-api-mesh/pull/137/commits/74cc4e3f7c4bf8f739ecad70dbad4c92f1fe730d)

File: src/utils.js [hunk 4]
```diff
@@ -458,10 +462,37 @@ async function parseSecrets(secretsContent) {
 		};
 		const compiledSecretsFileContent = parseEnv(secretsContent, envParserConfig);
 		const parsedSecrets = YAML.parse(compiledSecretsFileContent);
-		return parsedSecrets;
+		//check if secrets file is empty
+		if (!parsedSecrets) {
+			throw new Error(
+				chalk.red('Please check if your secrets YAML file is not empty and is valid.'),
```
[the remark this answers is not in the document]
Comment (nvyasadobe) [reply src/utils.js]: Done

File: src/utils.js [hunk 5]
```diff
@@ -458,10 +462,37 @@ async function parseSecrets(secretsContent) {
...
+			);
+		}
+		//check if parsedSecrets is string and not in k:v pair
+		if (typeof parsedSecrets === 'string') {
+			throw new Error(
+				chalk.red('Please provide a valid YAML in key:value format and not as string.'),
```
[the remark this answers is not in the document]
Comment (nvyasadobe) [reply src/utils.js]: DONE

File: src/utils.js [hunk 6]
```diff
@@ -458,10 +462,37 @@ async function parseSecrets(secretsContent) {
...
+}
+
+/**
+ * This function returns user friendly errors that occurs while YAML.parse
+ *
+ * @param error errors from YAML.parse
+ */
+function getSecretsYamlParseError(error) {
+	if (error.code === 'BAD_INDENT') {
+		return 'Bad indentation error occurred while parsing YAML data: ' + error.message;
```
[the remark this answers is not in the document]
Comment (nvyasadobe) [reply src/utils.js]: DONE

File: src/utils.js (hunk 2 above)
Comment (nvyasadobe) [reply src/utils.js]: DONE

File: src/utils.js (hunk 3 above)
Comment (nvyasadobe) [reply src/utils.js]: DONE

**Review by revanth0212 — APPROVED** [review]

**Pull request merged**